### PR TITLE
Upgrade to Go1.18

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Updates everything to Go1.18.5.

### Describe how to test/QA your changes

General agent QA -- at this stage, running in various clusters and looking for issues.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by maycmlee — APPROVED** [review]
👍 for docs.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/DataDog/datadog-agent
 
-go 1.17
+go 1.18
```
Comment (paulcacheux) [inline go.mod:3]: You also need to update the `.go-version` at the root, with `1.18.5`

Comment (djmitche) [reply go.mod:3]: Huh, I think that got reverted somehow.

**Comment by djmitche** [comment]
I didn't realize I had forgotten to mark this as "draft" - sorry!

**Comment by djmitche** [comment]
It looks like the only two non-green things here were due to GitHub workers dying, not to Go1.18.  One is CodeQL, which shouldn't depend on the Go version.  The other is the MacOS tests, which have secretly been running Go1.18 for a day or two already without trouble.  At any rate, I've re-run them both.  Now it's time for review.

**Review by paulcacheux — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Comment by paulcacheux** [comment]
Great work thanks ! Do you plan to setup a go workspace afterwards ? https://go.dev/doc/tutorial/workspaces

**Comment by djmitche** [comment]
> Great work thanks ! Do you plan to setup a go workspace afterwards ? https://go.dev/doc/tutorial/workspaces

I wasn't planning on doing so myself.  However, we've been discussing having each component bundle be a distinct module, in which case go workspaces might be very useful.  Are you interested in digging into that idea?

**Review by gh123man — APPROVED** [review]

**Review by amenasria — APPROVED** [review]

**Review by BaptisteFoy — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]

**Review by ppknap — APPROVED** [review]

**Review by ajgajg1134 — APPROVED** [review]

**Review by ahmed-mez — APPROVED** [review]

File: pkg/trace/agent/fuzz_test.go [hunk 2]
```diff
@@ -3,7 +3,7 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-//go:build go1.18
+//go:build go1.18 && disabled
```
Comment (ahmed-mez) [inline pkg/trace/agent/fuzz_test.go:6]:
```suggestion
//go:build go1.18 && !windows
```
As a workaround to https://github.com/golang/go/issues/52268 - the [fix](https://github.com/golang/go/commit/c3470ca83cfdbcc71ce162b00d6b9a0004a7f649) is in 1.19

**Pull request merged**